spinalcordtoolbox/utils/shell.py

```diff
@@ -98,6 +99,38 @@ def display_viewer_syntax(files, colormaps=[], minmax=[], opacities=[], mode='',
         printv(cmd + '\n', verbose=1, type='info')
 
 
+class SCTArgumentParser(argparse.ArgumentParser):
+    """Parser that centralizes initialization steps common across all SCT scripts."""
+    def __init__(self, *args, **kwargs):
+        def update_parent_default(key, value):
+            """A polite way of letting a child class have different default values than the parent class."""
+            # Source: https://stackoverflow.com/a/41623488
+            argspec = inspect.getfullargspec(super(SCTArgumentParser, self).__init__)
+            arg_index = argspec.args.index(key)
+            if len(args) < arg_index and key not in kwargs:
+                kwargs[key] = value
+
+        update_parent_default('formatter_class', SmartFormatter)
+
+        # Update "usage:" message to match how SCT scripts are actually called (no '.py')
+        frame = inspect.stack()[1]
+        module = inspect.getmodule(frame[0])
+        update_parent_default('prog', os.path.basename(module.__file__).strip(".py"))
+
+        # Disable "add_help", because it won't properly add '-h' to our custom argument groups
+        # (We use custom argument groups because of https://stackoverflow.com/a/24181138)
+        update_parent_default('add_help', False)
+
+        super(SCTArgumentParser, self).__init__(*args, **kwargs)
+
+    def error(self, message):
+        """Overridden parent method. Ensures that calling script with no args prints the help. See issue #3137."""
```

Comment:
Ahh, good point. I've been doing this in a lot of places, because it's helpful for keeping comments one-liners, but one-liners don't _reeeeally_ matter in this context. So, this is probably a habit I should update. :stuck_out_tongue_closed_eyes: